### PR TITLE
Port proximal adagrad optimizer from tf.train.ProximalAdagradOptimizer

### LINKED ISSUE #591: Add Keras implementation for ProximalAdagradOptimizer
**Describe the feature and the current behavior/state.**
ProximalAdagradOptimizer is currently implemented in TF 1.x as [`tf.train.ProximalAdagradOptimizer`.](https://github.com/tensorflow/tensorflow/blob/r1.14/tensorflow/python/training/proximal_adagrad.py)
As its API is deprecated in TF 2.0 and users are encouraged to use `tf.keras.optimizers.*` in TF 2.0, we need an equivalent Keras-version implementation for it. Besides, `ProximalAdagradOptimizer` is supported in [CannedEstimatorV1,](https://github.com/tensorflow/estimator/blob/3a35a2b01a5831cf7b22fa6226e7547d2e9d0465/tensorflow_estimator/python/estimator/canned/dnn_linear_combined.py#L438) but not yet supported in the V2 version due to the lack of its Keras version. It should be prioritized to fill the feature gap asap.

**Relevant information**
- Are you willing to contribute it (yes/no): 
Sorry but I don't have the bandwith now. 
- Are you willing to maintain it going forward? (yes/no):
Sure, I can help on this.
- Is there a relevant academic paper? (if so, where):
Yes. Check this [paper](http://papers.nips.cc/paper/3793-efficient-learning-using-forward-backward-splitting.pdf).
- Is there already an implementation in another framework? (if so, where):
Yes. It's already implemented in tensorflow [here](https://github.com/tensorflow/tensorflow/blob/r1.14/tensorflow/python/training/proximal_adagrad.py).
- Was it part of tf.contrib? (if so, where):
No.

**Which API type would this fall under (layer, metric, optimizer, etc.)**
optimizer
**Who will benefit with this feature?**
All users that use `tf.train.ProximalAdagradOptimizer` in 1.x and want to migrate to 2.0.
**Any other info.**

### PR DESCRIPTION
**Pull request by WindQAQ**
Closes #591. I am currently using `tensorflow.python.training.*`. Not sure if core TensorFlow will clean up unused APIs in the future.

Also, CC kernels for proximal adagrad are only registered for CPU/XLA, so I wonder if we should fork CC kernels in addons and make it compatible with GPU device. Test coverage is not enough, but want to discuss the situation first.
```
Registered:  device='XLA_GPU'; T in [DT_FLOAT, DT_DOUBLE, DT_BFLOAT16, DT_HALF]
  device='XLA_CPU'; T in [DT_FLOAT, DT_DOUBLE, DT_BFLOAT16, DT_HALF]
  device='XLA_CPU_JIT'; T in [DT_FLOAT, DT_DOUBLE, DT_BFLOAT16, DT_HALF]
  device='XLA_GPU_JIT'; T in [DT_FLOAT, DT_DOUBLE, DT_BFLOAT16, DT_HALF]
  device='CPU'; T in [DT_DOUBLE]
  device='CPU'; T in [DT_FLOAT]
```
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/kernels/training_ops.cc#L1562

### CODE REVIEW

**Comment by WindQAQ** [comment]
@yhliang2018, Hi Yanhui, could you review this if time allows? Thank you!

**Comment by gabrieldemarmiesse** [comment]
@googlebot I consent

**Comment by gabrieldemarmiesse** [comment]
@WindQAQ  I've merged master into your branch to update it and fixed any formatting/conflicts it might have. If you need to do some more modifications, please do `git pull` beforehand.

**Comment by gabrieldemarmiesse** [comment]
@WindQAQ  should we close this pull request?
